projects/plugins/jetpack/modules/subscriptions.php

```diff
@@ -1034,3 +1035,49 @@ public function register_post_meta() {
 Jetpack_Subscriptions::init();
 
 require __DIR__ . '/subscriptions/views.php';
+
+/**
+ * Gate access to posts
+ *
+ * @param string $the_content Post content.
+ *
+ * @return string
+ */
+function check_for_paid_subscription( $the_content ) {
+
+	require_once __DIR__ . '/../extensions/blocks/premium-content/_inc/subscription-service/include.php';
+	require_once __DIR__ . '/../modules/memberships/class-jetpack-memberships.php';
+
+	$protection_level = get_post_meta( get_the_ID(), \Automattic\Jetpack\Extensions\Subscriptions\META_NAME_FOR_POST_LEVEL_ACCESS_SETTINGS, true );
+
+	$can_view = true;
```

Comment:
Should it just return on the if statement?

Reply:
Returning early will require more code since we want to return the `do_blocks` code if the user doesn't have access